packages/uicore/src/components/RadioButton/RadioButton.tsx

```diff
@@ -48,7 +50,8 @@ export function RadioButton({
         type="radio"
         name={name}
         value={value}
-        defaultChecked={checked}
+        defaultChecked={defaultChecked}
```

Comment:
snapshots in harness-core-ui may be fail after this change

Reply:
@avinashmadhwani02 as pinged in the slack thread, 
there won't be snapshot failures everywhere were we use this component, because in html, the prop will be checked instead of default-checked or anything else.
Snapshot failures that may occur due to behaviour changes will occur only in components that have a functioning bug, which I believe are very less, we can update their snapshots.

is it okay? or should we got a different route?